scanner/enricher/csaf/csaf.go

```diff
@@ -0,0 +1,221 @@
+		u = cfg.URL
+		if !strings.HasSuffix(u, "/") {
+			u += "/"
+		}
+	}
+	var err error
+	e.base, err = url.Parse(u)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Name implements driver.Enricher and driver.EnrichmentUpdater.
+func (*Enricher) Name() string {
+	return "stackrox.rhel-csaf"
+}
+
+// Enrich implements driver.Enricher.
+func (e *Enricher) Enrich(ctx context.Context, g driver.EnrichmentGetter, r *claircore.VulnerabilityReport) (string, []json.RawMessage, error) {
+	ctx = zlog.ContextWithValues(ctx, "component", "enricher/csaf/Enricher/Enrich")
+
+	m := make(map[string][]json.RawMessage)
+
+	erCache := make(map[string][]driver.EnrichmentRecord)
+	for id, v := range r.Vulnerabilities {
+		vulnName := vulnerabilityName(v)
+		ctx := zlog.ContextWithValues(ctx, "original_vuln", v.Name, "vuln", vulnName)
+		rec, ok := erCache[vulnName]
+		if !ok {
+			ts := []string{vulnName}
+			var err error
+			rec, err = g.GetEnrichment(ctx, ts)
+			if err != nil {
+				return "", nil, err
+			}
+			erCache[vulnName] = rec
+		}
+		zlog.Debug(ctx).
+			Int("count", len(rec)).
+			Msg("found records")
+		for _, r := range rec {
+			m[id] = append(m[id], r.Enrichment)
+		}
+	}
+	if len(m) == 0 {
+		return Type, nil, nil
+	}
+	b, err := json.Marshal(m)
+	if err != nil {
+		return Type, nil, err
+	}
+	return Type, []json.RawMessage{b}, nil
+}
+
+var (
+	// The following vars match pkg/scannerv4/mappers/mappers.go and are copied here to prevent
+	// circular dependencies.
+
+	// Updater patterns are used to determine the security updater the
+	// vulnerability was detected.
+
+	rhelUpdaterName = (*vex.Updater)(nil).Name()
+
+	// Name patterns are regexes to match against vulnerability fields to
+	// extract their name according to their updater.
+
+	// cveIDPattern captures CVEs.
+	cveIDPattern = regexp.MustCompile(`CVE-\d{4}-\d+`)
+	// rhelVulnNamePattern captures known Red Hat advisory patterns.
+	// TODO(ROX-26672): Remove this and show CVE as the vulnerability name.
+	rhelVulnNamePattern = regexp.MustCompile(`(RHSA|RHBA|RHEA)-\d{4}:\d+`)
+
+	// vulnNamePatterns is a default prioritized list of regexes to match
+	// vulnerability names.
+	vulnNamePatterns = []*regexp.Regexp{
+		// CVE
+		cveIDPattern,
+		// GHSA, see: https://github.com/github/advisory-database#ghsa-ids
+		regexp.MustCompile(`GHSA(-[2-9cfghjmpqrvwx]{4}){3}`),
+		// Catchall
+		regexp.MustCompile(`[A-Z]+-\d{4}[-:]\d+`),
+	}
+)
+
+// vulnerabilityName searches the best known candidate for the vulnerability name
+// in the vulnerability details. It works by matching data against well-known
+// name patterns, and defaults to the original name if nothing is found.
+//
+// TODO: This is modified from pkg/scannerv4/mappers/mappers.go to prevent circular dependencies.
+// We should either combine these two, or better yet, remove the need for these.
+// Any changes done here should be considered for the source, too.
+func vulnerabilityName(vuln *claircore.Vulnerability) string {
```

Comment:
Now I see what you meant in that comment. Not sure if it was only me or other people will also get confused if/when reading that piece of code.